doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1261 @@
+    - (a) Type-2 update from Vtep-4 is processed and FRR converts the FDB entry in kernel from NUD_REACHABLE to NUD_NOARP(static) with NFEA_ACTIVITY_NOTIFY set.
+    - (b) Fdbsyncd receives the state change event from kernel and populates VXLAN_FDB_TABLE with ifname=PortChannel1, type=dynamic.
+    - (c) Fdborch observes that STATE_FDB_TABLE entry is already present. It updates the bitmap in the FDB cache as Local + Remote. And entry in the HW is not updated and remains as dynamic.
+4. MAC ages out on Vtep-1:
+    - (a) L2 table delete event from SAI is processed in Fdborch.
+    - (b) Fdborch removes STATE_FBD_TABLE entry, and resets Local flag in FDB cache. It observes that Remote flag is set in FDB cache, so it installs MAC back into the HW as static with ALLOW_MAC_MOVE flag.
+    - (c) Fdbsyncd receives STATE_FDB_TABLE delete event. It removes FDB entry from the kernel (or resets activity bit, if possible.)
+    - (d) FRR receives delete FDB notification (or with inactive_bit set.) BGP withdraws Type-2 update.
+5. Type-2 withdrawal is received on Vtep-4:
+    - (a) FRR updates the kernel FDB entry with IN_TIMER flag and starts hold-timer.
+    - (c) FRR should remove the FDB entry from the kernel for MAC to be learnt locally.
+    - (d) Fdborch receives the FDB delete event from the kernel and removes MAC from VXLAN_FDB_TABLE.
+6. MAC hold-timer expires on Vtep-4:
+    - (a) If there is no traffic hitting neither Vtep-1 nor Vtep-4, the MAC hold-timer will expire on Vtep-4.
+    - (b) FRR withdraws MAC advertisement with Proxy=1.
+    - (c) Vtep-1 receives MAC withdrawal and removes MAC from the kernel.
+    - (d) Fdbsyncd receives FDB delete event from the kernel and removes MAC from VXLAN_FDB_TABLE.
+    - (e) MAC is removed from HW on Vtep-1.
+
+### 2.2.6 Static Anycast Gateway
+Static Anycast Gateway (SAG) will be the primary and only mechanism for achieving active-active L3 gateway on the EVPN MH enabled VTEPs.
+
+In contrast to MCLAG, SVI interface MAC addresses are not synced between EVPN MH VTEPs. In order to configure same SVI IP address on the EVPN MH VTEPs for active-active L3 gateway, SAG IP addresses are required to be configured. However, if SVI interfaces on the EVPN MH VTEPs are required to have unique IP addresses (e.g. in case of BGP over EVPN MH), configure the unique SVI IP addresses on the EVPN MH VTEPs as usual.
+
+### 2.2.7 Mutual exclusion of MCLAG and EVPN MH configurations
+The MCLAG configuration will not be allowed if ESI is configured on one or more interface(s).
+
+### 2.2.8 ARP/ND suppression
+ARP/ND suppression will be supported in EVPN Multihoming scenarios. The VTEP will respond to the ARP/ND requests received on local access ports only for the ARP/ND installed against remote VTEPs. ARP/ND response will not be  generated by the VTEP if ARP/ND is installed on the local Multihomed ES, even though the ARP/ND learning happened on the remote VTEP multihoming a given ES.
```

Comment:
Similar to my previous question, initially, VTEP1 learns ARP. But later, the ARP packets goes to VTEP4 due to some hash changing locally on the host. In this case, how would VTEP4 handle these ARP packets?

Reply:
ARP rides on mac. If the mac is locally active on vtep-4, the corresponding type-2 routes will be advertised with proxy=0.

Reply:
After VTEP1 learns ARPs, it would use BGP to send this type 2 route to VTEP4. How does VTEP4 install this ARP entry to kernel? since this ARP entry should not be timeout unless VTEP1 withdrew it.

Once VTEP4 learns the same ARP entry locally, would kernel be able to flip previous remote learnt ARP entry to local entry? If the local ARP time sout later, which module detects it and reinstalls remote learnt ARP?